Source/WebCore/platform/ThreadGlobalData.cpp

```diff
@@ -76,42 +63,42 @@ void ThreadGlobalData::setWebCoreThreadData()
     ASSERT(&threadGlobalData() != sharedMainThreadStaticData);
 
     // Set WebThread's ThreadGlobalData object to be the same as the main UI thread.
-    **staticData = adoptRef(sharedMainThreadStaticData);
+    Thread::current().m_applicationData = adoptRef(sharedMainThreadStaticData);
```

Comment:
`ASSERT(&threadGlobalData() != sharedMainThreadStaticData);` implies that `Thread::current().m_applicationData` should already be pointing to `sharedMainThreadStaticData`.  Do we still need this `ThreadGlobalData::setWebCoreThreadData` function?

Maybe remove this assignment, and rename this function to `assertWebCoreThreadData()` instead?

Reply:
Ah, no. `ASSERT(&threadGlobalData() != sharedMainThreadStaticData);` means that `Thread::current().m_applicationData` is not pointing to `sharedMainThreadStaticData`.